### PR TITLE
Update to 2.3.8 and build for Python 3.12

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
werkzeug 2.3.8

**Destination channel:** defaults

### Links

- [PKG-3222](https://anaconda.atlassian.net/browse/PKG-3222)
- [Upstream repository](https://github.com/pallets/werkzeug/tree/2.3.8)
- [Upstream changelog/diff](https://github.com/pallets/werkzeug/compare/2.2.3..2.3.8)

### Explanation of changes:

* Update to 2.3.8
* Build with Python 3.12

[PKG-3222]: https://anaconda.atlassian.net/browse/PKG-3222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
 requirements:
   host:
     - python
     - pip
-    - setuptools
-    - wheel
+    - flit-core
   run:
     - python
     - MarkupSafe >=2.1.1
```
Comment (cbouss) [inline recipe/meta.yaml:24]:
```suggestion
    - MarkupSafe >=2.1.1
  run_constrained:
    - watchdog >=2.3
```

**Review by cbouss — COMMENTED** [review]
Could we update instead to https://werkzeug.palletsprojects.com/en/3.0.x/changes/#version-2-3-8 ?

This has the vulnerability fix and avoids a version bump.

**Comment by JeanChristopheMorinPerso** [comment]
@cbouss done.

**Review by cbouss — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]
LGTM but it would be nice to test it with the downstream package `flask`

**Comment by JeanChristopheMorinPerso** [comment]
@skupr-anaconda ran the flask tests locally and everything looks good.

**Pull request merged**